### PR TITLE
Forbid multiple indexes on same field

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #16866

### CODE REVIEW

File: internal/rootcoord/meta_table.go [hunk 1]
```diff
@@ -1036,13 +1037,74 @@ func (mt *MetaTable) unlockIsSegmentIndexed(segID typeutil.UniqueID, fieldSchema
...
+						"creating multiple indexes on same field is not supported, "+
+							"collection: %s, field: %s, index name: %s, new index name: %s",
+						collMeta.GetSchema().GetName(), fieldSchema.GetName(),
+						info.GetIndexName(), idxInfo.GetIndexName())
+				}
+			}
+			// TODO: unexpected: what if index id not exist? Meta incomplete.
+			msg := fmt.Sprintf("index meta was incomplete, index id missing in indexID2Meta, collection: %d, field: %d, index_id: %d",
+				collMeta.GetID(), fieldSchema.GetFieldID(), f.GetIndexID())
+			log.Warn(msg)
```
Comment (xige-16) [inline internal/rootcoord/meta_table.go:1070]: If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.

File: internal/rootcoord/meta_table.go [hunk 2]
```diff
@@ -1036,13 +1037,74 @@ func (mt *MetaTable) unlockIsSegmentIndexed(segID typeutil.UniqueID, fieldSchema
 	return exist
 }
 
+func (mt *MetaTable) unlockGetCollectionInfo(collName string) (pb.CollectionInfo, error) {
+	collID, ok := mt.collName2ID[collName]
+	if !ok {
+		collID, ok = mt.collAlias2ID[collName]
+		if !ok {
+			return pb.CollectionInfo{}, fmt.Errorf("collection not found: %s", collName)
```
Comment (xige-16) [inline internal/rootcoord/meta_table.go:1045]: Is it better toi print error info as fmt.Errorf("collection  %s not found", collName) ?

File: internal/rootcoord/meta_table.go (hunk 1 above)
Comment (xige-16) [reply internal/rootcoord/meta_table.go:1070]: This information will be returned to the sdk, and the index id in the error info will not be easy to understand, so it is best to replace it with name

Comment (boyjqlong) [reply internal/rootcoord/meta_table.go:1070]: > This information will be returned to the sdk, and the index id in the error info will not be easy to understand, so it is best to replace it with name

Agree.

Comment (boyjqlong) [reply internal/rootcoord/meta_table.go:1070]: > If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.

`mt.indexID2Meta[f.GetIndexID()]` exist and `idxInfo.GetIndexName() == info.GetIndexName()` won't cause this function return error.

File: internal/rootcoord/meta_table.go (hunk 2 above)
Comment (boyjqlong) [reply internal/rootcoord/meta_table.go:1045]: Both way make sense to me. But I think maybe `"collection %s not found", collName` is more difficult for me to locate the code.

File: internal/rootcoord/meta_table.go (hunk 1 above)
Comment (boyjqlong) [reply internal/rootcoord/meta_table.go:1070]: > > If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.
> 
> `mt.indexID2Meta[f.GetIndexID()]` exist and `idxInfo.GetIndexName() == info.GetIndexName()` won't cause this function return error.

Sorry, my mistake, please ignore this.

**Pull request merged**